File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import *

def solution(participant, completion):
    p_dit=Counter(participant)
    c_dit=Counter(completion)
    
    for person in participant:
        if c_dit[person]!=p_dit[person]:
            answer=person
            
    return answer


#배열 participant , completion 완주 못한 선수 return 

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> map = new HashMap<>();
        
        for (int j=0; j<completion.length; j++) {
            if (map.containsKey(completion[j])) {
                 map.put(completion[j], map.get(completion[j])+1);
            }
            else map.put(completion[j],1);
        }
        
        for (int i=0; i<participant.length; i++) {
            if (map.containsKey(participant[i])) {
                if (map.get(participant[i])==0) {
                    answer = participant[i];
                    break;  
                }
                else {
                map.put(participant[i], map.get(participant[i])-1);
                }
            }
            else  {
                answer = participant[i];
                break; } 
            
        }
        return answer;
    }
}